pyphare/pyphare/pharesee/hierarchy/patchlevel.py

```
@@ -13,3 +18,46 @@ def level_range(self):
+        return type(self) is type(that) and len(self.patches) == len(that.patches)
+
+
+def patch_level_array_ufunc(patch_level, ufunc, method, *inputs, **kwargs):
+    if method != "__call__":
+        return NotImplemented
+
+    if not all([patch_level.is_compatible(o) for o in inputs]):
+        raise TypeError("PatchLevel: incompatible arguments")
+
+    ps = [x.patches for x in inputs]
+    out = [getattr(ufunc, method)(*p, **kwargs) for p in zip(*ps)]
```

Ruff (0.15.9)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)